server/routerlicious/packages/lambdas/src/alfred/index.ts

```diff
@@ -622,18 +623,19 @@ export function configureWebSocketServices(
 				"broadcastSignal",
 				// eslint-disable-next-line @typescript-eslint/no-misused-promises
 				async (broadcastSignal: IBroadcastSignalEventPayload) => {
-					const { signalRoom, signalContent } = broadcastSignal;
-
-					// No-op if the room (collab session) that signal came in from is different
-					// than the current room. We reuse websockets so there could be multiple rooms
-					// that we are sending the signal to, and we don't want to do that.
-					if (
-						signalRoom.documentId === room.documentId &&
-						signalRoom.tenantId === room.tenantId
-					) {
-						try {
-							const contents = JSON.parse(signalContent) as IRuntimeSignalEnvelope;
-							const runtimeMessage = createRuntimeMessage(contents);
+					try {
+						const signalRoom = broadcastSignal.signalRoom;
+						const signalContent =
+							broadcastSignal.signalContent as IRuntimeSignalEnvelope;
```

Comment:
Nit: I generally recommend casting to `Partial<Foo>` whenever casting an object of unknown type. That way you're forced to handle the cases where the data you're looking for isn't there. Doesn't look like it would have a major impact on this use case, but figured I would call it out 🙂

Reply:
Also, what does it mean if we get data that isn't `IRuntimeSignalEnvelope`? Is that invalid (should we throw / log an error)? Or does it mean something else? Would probably be worth documenting.

Reply:
> In this case would it be Partial?

Only as a means of guarding against malformed input. By typing it as `Partial<IRuntimeSignalEnvelope>`, the type system will force you to validate that the fields you care about are actually there.